EmpleoDotNet/Controllers/JobOpportunityController.cs

```diff
@@ -27,27 +28,18 @@ public ActionResult Index(JobOpportunityPagingParameter model)
         // GET: /JobOpportunity/Detail/4-jobtitle         
         public ActionResult Detail(string id)
         {
-            if (string.IsNullOrEmpty(id))
-                return RedirectToAction("Index");
-
-            var url = id.Split('-');
-
-            if (url.Length == 0)
-                return RedirectToAction("Index");
-
-            var value = 0;
-
-            if (!int.TryParse(url[0], out value))
+            var value = GetIdFromTitle(id);
```

Comment:
Yes. Indeed it would be better. I previously tried to change parameter id to title (so i can name the value as id) and made the routes work awkwardly (/Jobs?id=1-test-job instead of /jobs/1-test-job)